drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -520,6 +520,18 @@ static int nrf5_tx(const struct device *dev,
 
 	LOG_DBG("Result: %d", nrf5_data.tx_result);
 
+#if NRF_802154_ENCRYPTION_ENABLED
+	/*
+	 * When frame encryption by the radio driver is enabled, the frame stored in the tx_psdu
```

Comment:
Maybe update this comment?

```
	/*
	 * When frame encryption by the radio driver is enabled,
	 * the frame stored in the tx_psdu buffer is authenticated
	 * and encrypted in place. This frame must be propagated back
	 * to the upper layer for keeping the link frame counter up
	 * to date and to be used for retransmissions.
	 */
```

Reply:
I don't think it is needed only for retransmission, it's in general for keeping the link frame couter up to date. I commented it more general:
`* Additionally, when this option is enabled, the radio driver updates frame counters which needs also be propagated back to the upper layer.`

Reply:
In my opinion the comment should describe two cases:

1. After an unsuccessful TX attempt, frame must be propagated back to the upper layer for retransmission
2. For unsecured frame, for keeping the link frame couter up to date, frame must be propagated back to the upper layer